src/main/java/com/appland/appmap/record/EventTemplateRegistry.java

```diff
@@ -85,8 +87,11 @@ public Event cloneEventTemplate(int templateId, String eventAction)
           event.addParameter(param);
         }
       }
-    } catch (ArrayIndexOutOfBoundsException e) {
-      throw new UnknownEventException(String.format("unknown template for ordinal %d", templateId));
+    } catch (IndexOutOfBoundsException e) {
+      if (Properties.DebugHooks) {
+        Logger.println(String.format("unknown template for ordinal %d", templateId));
```

Comment:
The advantage to this being a hard failure is that it might push the user towards this workaround: https://github.com/applandinc/appmap-java/wiki/Frequently-Asked-Questions#-How-do-I-fix-an-unknown-template-for-ordinal-error .

I wonder if there's some other way we can direct them towards it?

Reply:
Thanks Alan. I wasn't aware this wiki page existed. I'm likely going to move this back to an exception if it's only related to PowerMock.